libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -263,27 +236,16 @@ pub(super) fn merge_local_folder_manifest(
     // /!\ Extra attention should be paid here if we want to add new fields
     // /!\ with their own sync logic, as this optimization may shadow them!
 
-    let merged_need_sync = merged_children != remote.children || merged_parent != remote.parent;
-    let updated = if merged_need_sync {
-        timestamp
-    } else {
-        remote.updated
-    };
-    // TODO: prevent sync pattern is not handled here !
-    let merged_manifest = LocalFolderManifest {
-        need_sync: merged_need_sync,
-        parent: merged_parent,
-        children: merged_children,
-        updated,
-        base: remote,
-        // TODO: confinement points support not implemented yet !
-        local_confinement_points: HashSet::new(),
-        // TODO: confinement points support not implemented yet !
-        remote_confinement_points: HashSet::new(),
-        speculative: false,
+    local_from_remote.speculative = false;
+    local_from_remote.need_sync =
+        merged_children != local_from_remote.children || merged_parent != local_from_remote.parent;
+    if local_from_remote.need_sync {
+        local_from_remote.updated = timestamp;
```

Comment:
The original code was
```rust
let updated = if merged_need_sync {
        timestamp
    } else {
        remote.updated
    };
```
I guess this has changed the behavior, why ?
(and we are missing a test here if the behavior has changed without us noticing !)

Reply:
Can you be more precise on why you think so ?

For me it's the same behavior, where we update `updated`, `children` and `parent` only if `children` or `parent` are different from the merged result and the remote.

Reply:
> Can you be more precise on why you think so ?

the previous code:
```rust
let updated = if merged_need_sync {
        timestamp
    } else {
        remote.updated
    };
```
the new code:
```rust
    let local_from_remote = LocalFolderManifest::from_remote_with_local_context(
        remote,
        prevent_sync_pattern,
        local,
        timestamp,
    );
[...]
    if local_from_remote.need_sync {
       local_from_remote.updated = timestamp
   }
```

I guess `local_from_remote.updated` get the same value than `remote.update` when building `local_from_remote` with `LocalFolderManifest::from_remote_with_local_context`', however `timestamp` is also passed to this function...

In fact it's even worse than that when reading the function: `LocalFolderManifest::restore_local_confinement_points` uses `timestamp` to update `updated` field ONLY IF `local_confinement_points` or `remote_confinement_points` are set... which means in your current code, `merged_manifest.updated` get set with `timestamp` if it is not `need_sync` but has confinement points (while the expected behavior is to have `merged_manifest.updated` to get set with `remote.updated`)

In that regard, the first version of the code is much cleaner given you don't have to think about how `LocalFolderManifest::from_remote_with_local_context` works to understand the code, and the full reasoning on what will be the `updated` value is in a single place.

Reply:
Ok

Reply:
Sorry for the never-ending comment, but I think your current code is still affected:
```rust
 let merge_update = if merge_need_sync {
        timestamp
    } else {
        local_from_remote.updated
    };
```

As I've said, `remote.updated` is not strictly equivalent to `local_from_remote.updated`

My guess (I didn't test the code) is if `local_confinement_points` or `remote_confinement_points` are set but `need_sync == false`, then `merge_update == local_from_remote.updated == timestamp` (while the correct behavior would be `merged_update == remote.updated`

Reply:
Given the indication of `remote.updated`, I should use `local_from_remote.base.updated` ?

Reply:
Done